### ISSUE TITLE
Banned after one DM (401 Unauthorized)

### ISSUE THREAD

**Issue by and0** [Merubokkusu/discord-spam-bots#103]
Bot joins server fine but as soon as DM text spam is launched bot gets banned after just one successful message. The message is randomized and is single line and contains no emojis.

1. Configure `tokens.txt` with one token
2. Launch join server (works)
3. Launch DM Text Spam 

 - OS: OSX
 - Python 3.7

```
Started Text DM Spam
80xxxxxxxxx546
 [+] (<discum.messages.messages.Messages->createDM) Post -> https://discord.com/api/v8/users/@me/channels

 [+] (<discum.messages.messages.Messages->createDM) {'recipients': ['80xxxxxxxxx546']}

 [+] (<discum.messages.messages.Messages->createDM) Response <- {"message": "Cannot send messages to this user", "code": 50007}

Something went wrong (;3;) relaunching...
161xxxxxxxxxxxx09
 [+] (<discum.messages.messages.Messages->createDM) Post -> https://discord.com/api/v8/users/@me/channels

 [+] (<discum.messages.messages.Messages->createDM) {'recipients': ['161xxxxxxxxxxxx09']}

 [+] (<discum.messages.messages.Messages->createDM) Response <- {"id": "83xxxxxxxxxxx33", "type": 1, "last_message_id": null, "recipients": [{"id": "161859548540305409", "username": "xxxxxx", "avatar": "aeaxxxxxxxxxxxxxxxc12e", "discriminator": "3780", "public_flags": 0}]}

 [+] (<discum.messages.messages.Messages->sendMessage) Post -> https://discord.com/api/v8/channels/83xxxxxxxxxxx33/messages

 [+] (<discum.messages.messages.Messages->sendMessage) {'content': '<3<3 Hello this is a test <3 ~<3~^_ ^', 'tts': False, 'nonce': '830xxxxxxxxxx000'}

 [+] (<discum.messages.messages.Messages->sendMessage) Response <- {"id": "830xxxxxxxxx1504", "type": 0, "content": "<3<3 Hello this is a test <3 ~<3~^_ ^", "channel_id": "83xxxxxxxxx9233", "author": {"id": "805918878466965546", "username": "xxxxxxxx", "avatar": "aaf8xxxxxxxxxxxx7015", "discriminator": "5418", "public_flags": 0}, "attachments": [], "embeds": [], "mentions": [], "mention_roles": [], "pinned": false, "mention_everyone": false, "tts": false, "timestamp": "2021-04-11T05:37:58.715000+00:00", "edited_timestamp": null, "flags": 0, "nonce": "830xxxxxxxxx8000", "referenced_message": null}

Sent message to 1618xxxxxxxxxx409
Complete!
30xxxxxxxxxx11947
 [+] (<discum.messages.messages.Messages->createDM) Post -> https://discord.com/api/v8/users/@me/channels

 [+] (<discum.messages.messages.Messages->createDM) {'recipients': ['300xxxxxxxxx47']}

 [+] (<discum.messages.messages.Messages->createDM) Response <- {"message": "401: Unauthorized", "code": 0}

Something went wrong (;3;) relaunching...
```

**Comment by and0** [comment]
Update: When making my own account I am able to get a few messages before being met with this phone verification response.

```
{'message': 'You need to verify your account in order to perform this action.', 'code': 40002}
```

**Comment by and0** [comment]
Something about the process of `createDM` & `sendMessage` is getting flagged and I'm not sure what it is.. I've made sure to take all the precautionary measures such as using a different text string each time and also not being logged in to this account in another location as you've mentioned as well as using ample (30s) delay between each messages

**Comment by lowlypalace** [comment]
Same here, tried accounts with the verified phone number too. Using private proxies.

**Comment by Merubokkusu** [comment]
Can confirm that createDM is the cause of the ban, going to test some stuff out later today. A delay might fix it

**Comment by lowlypalace** [comment]
Cool, I also tried with 30 seconds delay but it still causes a ban.

**Comment by lowlypalace** [comment]
> Can confirm that createDM is the cause of the ban, going to test some stuff out later today. A delay might fix it

Cool, I also tried with 60 seconds delay but it still causes a ban.

**Comment by lowlypalace** [comment]
Any updates on this one? I also sent you a Twitter DM with some bugs I found @Merubokkusu

**Comment by anhvietgod** [comment]
> > Can confirm that createDM is the cause of the ban, going to test some stuff out later today. A delay might fix it
> 
> Cool, I also tried with 60 seconds delay but it still causes a ban.

How to modify the delay?

**Comment by anhvietgod** [comment]
Had the same problem! The bot seemed to work properly but got banned after 3 DMs, I tried to set the delay to 100 and used edge but still got a ban. I don't know if discord updated their algorithm or what else. I'm waiting for someone to fix it,  willing to donate!

**Comment by ghost** [comment]
Same here, only tried to join my server and..
![image](https://user-images.githubusercontent.com/79478046/118362615-7226de00-b590-11eb-8f27-0973872cfd83.png)
BOOM. 
![image](https://user-images.githubusercontent.com/79478046/118362627-7bb04600-b590-11eb-9b19-5b0b12ab979b.png)
Error 403 - forbidden.

**Comment by anhvietgod** [comment]
> Same here, only tried to join my server and..
> ![image](https://user-images.githubusercontent.com/79478046/118362615-7226de00-b590-11eb-8f27-0973872cfd83.png)
> BOOM.
> ![image](https://user-images.githubusercontent.com/79478046/118362627-7bb04600-b590-11eb-9b19-5b0b12ab979b.png)
> Error 403 - forbidden.

Yeah I found another solution. Xyliase tool is the best solution for this. You must pay for it, but I think it's worth

**Comment by ghost** [comment]
Okay guys, i've found a workaround. Sadly, **it's not avaible for python.** You can use node.js. Libraries such as discord.js combined with puppeteer for auto-joining server and auto-account creator, and discord.js for spamming the channels will work great.

**Comment by Mannotcool** [comment]
I just want to mention this, but using a VPN or logging in from a lot of accounts on one IP causes the "Verify by Phone" to appear. This seems like a new prompt that discord has implemented to stop spam bots, it kind of sucks for us trying to test spam bot protection.
